\documentclass[12pt]{article}

\usepackage{textcomp}
\usepackage[cm]{fullpage}
\usepackage{amsfonts}
\usepackage{verbatim}
\usepackage[english]{babel}
\usepackage{csquotes}
\usepackage{color}
\usepackage{setspace}
\usepackage{lscape}
\usepackage{indentfirst}
\usepackage[normalem]{ulem}
\usepackage{booktabs}
% \usepackage{nag}
\usepackage[backend=biber,style=authoryear,maxbibnames=9,maxcitenames=2,uniquelist=false,doi=false,url=false,eprint=false]{biblatex}
\usepackage{float}
\usepackage{latexsym}
\usepackage{hyperref}
\usepackage{url}
% \usepackage{html}
\usepackage{epsfig}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{array}
%\usepackage{mhchem}
\usepackage{ifthen}
\usepackage{caption}
\usepackage{xcolor}
\usepackage{amsthm}
\usepackage{amstext}
\usepackage{nicefrac}
\usepackage{algorithm}
\usepackage{algorithmic}
\usepackage[scientific-notation=true]{siunitx}
\usepackage{subfigure}
\usepackage[flushleft]{threeparttable}
\usepackage{lineno}
\usepackage{adjustbox}
\usepackage{ragged2e}
\usepackage{authblk}
\usepackage{multirow}
\usepackage{tcolorbox}
\usepackage{soul}

\setuldepth{after}
\setlength{\parskip}{1em}
\renewcommand{\baselinestretch}{2.0}
\renewcommand\Affilfont{\small}

\makeatletter
\def\verbatim@font{\linespread{1}\scriptsize\ttfamily}
\makeatother

\addbibresource{StarBEAST2-refs.bib}

\begin{document}

\title{StarBEAST2 v1.0.0 tutorial}
\author[1]{Huw A. Ogilvie\thanks{huw.a.ogilvie@rice.edu}}
\affil[1]{Department of Computer Science, Rice University, Houston, Texas}

\maketitle

\clearpage

\section{Introduction}
\label{sec:intro}

StarBEAST2 is a method for multilocus, multispecies coalescent phylogenetic
inference \parencite{Ogilvie2017}. Multilocus methods like StarBEAST2 take as
input one or more multiple sequence alignments derived from genomic loci.
There is assumed to be no recombination within each locus and free
recombination between loci. Therefore the sequence alignments should be
relatively short, and relatively distantly spaced along the genome. Genomic
data sets which are typically a good fit to the StarBEAST2 model include
RAD-seq loci \parencite{Ogilvie2016} or exons \parencite{Scornavacca2017}.

StarBEAST2 can be used to estimate species tree topologies and divergence
times, the topologies and coalescence times of gene trees, the substitution
model rates and base frequencies for those gene trees, per-species population
sizes and per-species molecular clock rates.

Newer versions of StarBEAST2 supports the integration of molecular sequences
and morphological characters for so-called ``total evidence'' analyses, and
the use of fossil data for tip-dating \parencite{Ogilvie2021}. Fossil data
can include ancient DNA, morphological characters, and the estimated ages of
fossils. Tip-dating calibrates the trees in absolute time, typically millions
of years.

This tutorial will cover using StarBEAST2 to estimate a species tree including
per-species population sizes of the genus \textit{Canis} from exon sequences
(Section~\ref{sec:speciesTree}),
to estimate per-species molecular clock rates
(Section~\ref{sec:relaxedClock}),
and to conduct a total evidence tip-dating study
(Section~\ref{sec:FBD}).

\section{Preliminaries}
\label{sec:prelim}

Before doing anything, we need to install StarBEAST2, which is available as a package for
the phylogenetic software platform BEAST2. This tutorial is for version 2.7 of the platform, which you can download from \url{http://beast2.org}.
Make sure you have the latest update to BEAST2 version 2.7 before proceeding.

After downloading and installing BEAST2 on your computer, open BEAUti ---
the graphical user interface for BEAST2. To install StarBEAST2 (or
any BEAST2 package), open the File menu and select Manage Packages (Figure~\ref{fig:managePackages}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/managePackages.png}
\caption
{Opening the package manager}
\label{fig:managePackages}
\end{figure}

\clearpage

Select StarBEAST2 from the list of available packages and click the install button, which will take
care of the installation for you (Figure~\ref{fig:installStarBEAST2}). After the installation of StarBEAST2 is finished, just like for any BEAST2 package, you
\textbf{must} quit and relaunch BEAUti.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/installStarBEAST2.png}
\caption
{Installing StarBEAST2}
\label{fig:installStarBEAST2}
\end{figure}

After relaunching BEAUti, you will notice four new templates (Figure~\ref{fig:strictClockTemplate}). The first
is ``StarBeast2'', which is for a strict molecular clock or relaxed
gene tree clocks uncorrelated with the species tree lineages. The others
are ``SpeciesTreeUCLN'', ``SpeciesTreeRLC'' and ``SpeciesTreeUCED'' which
enable per-species clock rates, as described by \cite{Ogilvie2016}.
The relaxed clock models used by those templates are uncorrelated log-normal (UCLN),
relaxed local clock (RLC), and uncorrelated exponential distribution (UCED)
respectively.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/strictClockTemplate.png}
\caption
{Selecting a StarBEAST2 template}
\label{fig:strictClockTemplate}
\end{figure}

To visualize posterior distributions of trees, we will be using DensiTree
\parencite{Bouckaert2010} which is included with BEAST2. We will also use the web application IcyTree to view summary trees, which is available at \url{http://icytree.org/}. To analyze continuous parameters we will use Tracer, which can be downloaded from \url{https://beast.community/tracer}.

\section{Reconstructing the species tree of \textit{Canis}}
\label{sec:speciesTree}

The genus \textit{Canis} includes species such as \textit{Canis lupus} (wolves),
\textit{Canis latrans} (coyotes), and \textit{Canis dirus} (dire wolves, extinct).
Because the
taxonomy of this genus is quite messy, it also includes
\textit{Cuon alpinus} and \textit{Lycaon pictus}.
In this section we will reconstruct the species tree of \textit{Canis} using StarBEAST2.

\subsection{Importing alignments}
\label{subsec:importingAlignments}

First up, create a new folder somewhere with a sensible name like
``CanisPhylogeny''. Relaunch BEAUti and load up the strict clock
template by selecting ``StarBeast2'' from the Templates submenu of the File menu
(Figure~\ref{fig:strictClockTemplate}). Now we will import multiple sequence
alignments of the 16 nuclear loci sequenced by \cite{LindbladToh2005}. Select
Import Alignments from the File menu, and navigate to the ``data'' subfolder
of the tutorial. Select all 16 FASTA files, and click OK to import
(Figure~\ref{fig:fastaFileImport}). Do not select ``morphology-canis.nexus''
which is for the total evidence study.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/fastaFileImport.png}
\caption
{Importing \textit{Canis} multiple sequence alignments}
\label{fig:fastaFileImport}
\end{figure}

After you click OK, you will be asked what type of sequences are in the files.
As all 16 files are MSAs of nucleotide sequences, select ``all are nucleotide''
and click OK again to continue. There should now be 16 partitions listed in
BEAUti.

\subsection{Linking models}
\label{subsec:linkingModels}

As a general rule, clock models should be linked when using the strict clock StarBEAST2 template. Rates between
loci are still allowed to vary and the mean rate among all sites is fixed at 1. Therefore
by linking the clocks, the average clock rate can be fixed at an \textit{a priori} value
to calibrate the analysis, or estimated when node-dating or tip-dating is used.
Select all 16 loci and click the ``Link Clock Models'' button. Now all loci
should have ``APOBS1'' as the clock model name. The site model, clock model and tree names are text fields that you can click on to edit. Change the name of the clock model to ``molecular'' and it should look like Figure~\ref{fig:linkModels}.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/linkModels.png}
\caption
{Linking clock models}
\label{fig:linkModels}
\end{figure}{}

\begin{tcolorbox}[colback=green!5,colframe=green!40!black,title=Mixing nuclear and mitochondrial loci]
StarBEAST2 can be used with a combination of mitochondrial and nuclear loci, but this requires careful thought when specifying the model. One possibility is to link the nuclear loci as above, but not to link the mitochondrial loci. Then when editing the site models, untick ``estimate'' for the substitution rate of each mitochondrial locus. Now separate clock rates will be used for each mitochondrial locus, in addition to the overall clock rate which applies to nuclear loci.

If the species tree is calibrated using an \textit{a priori} nuclear or mitochondrial molecular clock rate, you can set that rate in the Clock Model panel. If the mitochondrial rate is fixed, the nuclear rate should probably be estimated and \textit{vice versa}. If the species tree is calibrated using node or tip dating, then all clock rates should probably be estimated.

Make sure appropriate priors are used for all estimated clock rates. This could be a $1/X$ prior for rates where you genuinely have no prior knowledge. However in most circumstances you will have a general idea of the clock rate, e.g. within an order of magnitude. In that case, you can set a broad prior with a mean based on your prior knowledge. Broad priors include Log Normal with a large standard deviation (up to about 2), or a Gamma distribution with a small $\alpha$ (set ``alpha'' to 1 or 2).
\end{tcolorbox}

\begin{tcolorbox}[colback=purple!5,colframe=purple!40!black,title=Linking site models]
When clock models are linked as is recommended by this tutorial, linking site models will result in every gene having exactly the same
clock rate. As this is unrealistic, generally site models should \textbf{not}
be linked using BEAUti when setting up a StarBEAST2 analysis.
\end{tcolorbox}

\subsection{Specifying species names}
\label{subsec:speciesNames}

In the FASTA files, each sequence has a name like ``Canis\_anthus\_a''.
Here \textit{Canis anthus} is the binomial name, and ``a'' is a haplotype.
For the data supplied with this tutorial, each species has two haplotypes ``a'' and ``b'', both from
the same diploid individual. However in other studies multiple individuals may
be sequenced, so an arbitrary number of haplotypes are available per species.

To assign haplotypes to species, select the Taxon Sets tab in BEAUti, then
click the Guess button. To assign the correct names, keep ``use everything''
selected, but change ``after first'' to ``before last''. Leave the underscore
in the text box and click OK (Figure~\ref{fig:guessTaxonMap}). This way everything before
the last underscore in the haplotype name will be used as the species name,
so ``Canis\_anthus'' will be the species name for ``Canis\_anthus\_a''.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/guessTaxonMap.png}
\caption
{Guessing species names from haplotype names}
\label{fig:guessTaxonMap}
\end{figure}

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/taxonMap.png}
\caption
{The assignment of haplotypes to species}
\label{fig:taxonMap}
\end{figure}

Now each haplotype (Taxon) should have a corresponding species (Species/Population).
Based on how we assigned the species names, there should be two haplotypes
``a'' and ``b'' for each species (Figure~\ref{fig:taxonMap}).

\subsection{Gene Ploidy and Population Model}
\label{subsec:ploidyAndPopModel}

Ignore the Tip Dates tab, which is for tip-dating and will be covered in
section~\ref{sec:FBD} of the tutorial. Open the Gene Ploidy tab, and observe that the default value
for all loci is 2.0. This is because there are two copies of a locus in each
individual for diploid populations, so we scale the effective population sizes
by 2.0. If you use any mitochondrial or Y/W chromosomal loci, you should
change their ploidy to 0.5, because there is effectively only half as many copies of those molecules as there are individuals (male mitochondrial genomes do not count, as they are not inherited by offspring). The ploidy of X/Z chromosomal loci should be set to 1.5 for the
same reason.

Select the Population Model tab. By default analytical integration is used,
which is slightly faster but does not produce estimates of per-species population sizes.
Change the model to ``Constant Populations'', which will add effective population
sizes to the species tree output (Figure~\ref{fig:constantPopulations}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/constantPopulations.png}
\caption
{Changing the population model}
\label{fig:constantPopulations}
\end{figure}

\clearpage

\subsection{Site Model}
\label{subsec:siteModel}

Select the Site Model tab, and you will see the site model for the first partition
displayed. Change ``JC69'' to ``HKY'', and then set the frequencies to
empirical (Figure~\ref{fig:hky}).
HKY is more flexible because it allows nucleotide transitions to have a different
rate relative to transversions \parencite{Hasegawa1985}.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/hky.png}
\caption
{Setting the site model to HKY with empirical base frequencies}
\label{fig:hky}
\end{figure}

Now to change the site model for \textit{all} loci to HKY with empirical base frequencies,
select all of the partitions in the left hand column with the shift key. Choose to clone the site model you just configured (presumably for APOBS1), then
click ``OK'' to apply the same site model used for the first locus
to everything else (Figure~\ref{fig:cloneSiteModel}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/cloneSiteModel.png}
\caption
{Cloning the site model}
\label{fig:cloneSiteModel}
\end{figure}

In a more serious analysis, you might consider setting the number of gamma
categories to 4. This allows for different sites to evolve at
different rates, an obviously more realistic model \parencite{Yang1994}. However
the phylogenetic likelihood must be calculated once for each category, so 4
categories will be be $4\times$ slower. That likelihood calculation is a major
part of the StarBEAST2 algorithm, so more gamma rate categories will require
more computer time.

\subsection{Clock Model}
\label{subsec:clockModel}

\cite{Hugall2007} estimated that the molecular
clock rate for the RAG-1 nuclear coding gene in mammals is approximately
$10^{-3}$ substitutions per site per million years. While we don't know
the average rate across all loci in our data within \textit{Canis}, we can
use it as a \textbf{very approximate} calibration. Open the clock model panel
and set the rate to ``0.001'', to match the \textit{a priori} estimate
(Figure~\ref{fig:strictClockModel}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/strictClockModel.png}
\caption
{Using an \textit{a priori} clock rate for calibration}
\label{fig:strictClockModel}
\end{figure}

You can also use the scientific notation shorthand, 1e-3, which is equivalent
to 0.001.

\subsection{Priors and MCMC}
\label{subsec:MCMC}

Open the Priors panel and change ``Yule Model'' to ``Birth Death Model''.
These models are identical, except the birth-death model allows for a
non-zero rate of extinction.

StarBEAST2 is an MCMC method. These kind of methods do better at estimating
the posterior distributions (of trees or other parameters) the longer they
are run, although after a point there are diminishing returns. The default
chain length in StarBEAST2 is 10 million states, but for this analysis we
need a bit more for good estimates; change the chain length to 40 million
(Figure~\ref{fig:chainLength})

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/chainLength.png}
\caption
{Setting a longer MCMC chain length of 40 million iterations}
\label{fig:chainLength}
\end{figure}

Save your model as an XML file -- in the folder you created before importing
alignments -- by clicking Save As in the File menu. Navigate to the folder
and give your XML file a name like ``CanisPhylogeny.xml''.

\subsection{Running BEAST}
\label{subsec:runningBEAST}

You can run BEAST from the command line or using the GUI. To run your XML
from the command line, first navigate to the folder you saved the XML file into.
Then run ``/path/to/beast/bin/beast CanisPhylogeny.xml''. Make sure to change
/path/to/beast to match the folder where BEAST is installed on your computer.

BEAST should take about 5 to 10 minutes to finish the chain. Sampled statistics
and various parameters will be saved to ``starbeast.log'', species trees to
``species.trees'', and separate gene tree files will be created for each locus. The command line
output should start off looking something like what follows:


\begin{verbatim}                   BEAST v2.7.5 Prerelease, 2002-2023
             Bayesian Evolutionary Analysis Sampling Trees
                       Designed and developed by
 Remco Bouckaert, Alexei J. Drummond, Andrew Rambaut & Marc A. Suchard
                                    
                   Centre for Computational Evolution
                         University of Auckland
                       r.bouckaert@auckland.ac.nz
                        alexei@cs.auckland.ac.nz
                                    
                   Institute of Evolutionary Biology
                        University of Edinburgh
                           a.rambaut@ed.ac.uk
                                    
                    David Geffen School of Medicine
                 University of California, Los Angeles
                           msuchard@ucla.edu
                                    
                      Downloads, Help & Resources:
                           http://beast2.org/
                                    
  Source code distributed under the GNU Lesser General Public License:
                   http://github.com/CompEvol/beast2
                                    
                           BEAST developers:
   Alex Alekseyenko, Trevor Bedford, Erik Bloomquist, Joseph Heled, 
 Sebastian Hoehna, Denise Kuehnert, Philippe Lemey, Wai Lok Sibon Li, 
Gerton Lunter, Sidney Markowitz, Vladimir Minin, Michael Defoin Platel, 
          Oliver Pybus, Tim Vaughan, Chieh-Hsi Wu, Walter Xie
                                    
                               Thanks to:
          Roald Forsberg, Beth Shapiro and Korbinian Strimmer

Random number seed: 1691867441007

File: Canis-Phylogeny.xml seed: 1691867441007 threads: 1
Loading package BEAST.app v2.7.5 SA v2.1.1 MM v1.2.1 ORC v1.1.2
starbeast2 v1.0.0 BEAST.base v2.7.5 BEASTLabs v2.0.1 starbeast3 v1.1.7\end{verbatim}

At the end of the output will be statistics describing the performance of all the MCMC operators. This will look something like below, except I needed to replace some of the output with ellipses so it would fit in this document:

\begin{verbatim}Operator                                                         ...  Tuning    #accept    #reject      Pr(m)  Pr(acc|m)
...
beast.base.inference.operator.UpDownOperator(clockUpDownOperator ... 0.80611      18039      52115    0.00174    0.25713 
ScaleOperator(TreeScaler.t:Ch14)                                 ... 0.81774      18037      51748    0.00174    0.25847 
ScaleOperator(TreeRootScaler.t:Ch14)                             ... 0.52148      14234      55718    0.00174    0.20348 
Uniform(UniformOperator.t:Ch14)                                  ...       -     121376     228556    0.00872    0.34686 
SubtreeSlide(SubtreeSlide.t:Ch14)                                ... 0.65112     117291     232591    0.00872    0.33523 
Exchange(Narrow.t:Ch14)                                          ...       -      21860     326864    0.00872    0.06269 
Exchange(Wide.t:Ch14)                                            ...       -        804     347965    0.00872    0.00231 
WilsonBalding(WilsonBalding.t:Ch14)                              ...       -       1394     347821    0.00872    0.00399 
beast.base.inference.operator.kernel.BactrianDeltaExchangeOperat ... 0.39628      86822     284324    0.00930    0.23393 
ScaleOperator(KappaScaler.s:APOBS1)                              ... 0.18949       6789      16652    0.00058    0.28962 
ScaleOperator(KappaScaler.s:APOBS2)                              ... 0.21016       7722      15778    0.00058    0.32860 
ScaleOperator(KappaScaler.s:BDNF)                                ... 0.18970       7620      15658    0.00058    0.32735 
ScaleOperator(KappaScaler.s:BRCA1S1)                             ... 0.18986       7245      15817    0.00058    0.31415 
ScaleOperator(KappaScaler.s:BRCA1S2)                             ... 0.19130       6763      16591    0.00058    0.28959 
ScaleOperator(KappaScaler.s:Ch14)                                ... 0.30988       6979      16290    0.00058    0.29993 
ScaleOperator(KappaScaler.s:Ch21)                                ... 0.25278       6882      16419    0.00058    0.29535 
ScaleOperator(KappaScaler.s:Ch24)                                ... 0.23802       7303      16019    0.00058    0.31314 
ScaleOperator(KappaScaler.s:CHST12)                              ... 0.18101       7118      16184    0.00058    0.30547 
ScaleOperator(KappaScaler.s:CMKOR1)                              ... 0.18940       7070      16249    0.00058    0.30319 
ScaleOperator(KappaScaler.s:FGFR3)                               ... 0.20670       7454      15770    0.00058    0.32096 
ScaleOperator(KappaScaler.s:GHRex09)                             ... 0.19213       7583      15839    0.00058    0.32376 
ScaleOperator(KappaScaler.s:RAG1)                                ... 0.22845       7414      15823    0.00058    0.31906 
ScaleOperator(KappaScaler.s:TMEM20)                              ... 0.21491       7438      15926    0.00058    0.31835 
ScaleOperator(KappaScaler.s:VANGL2)                              ... 0.19990       7590      15886    0.00058    0.32331 
ScaleOperator(KappaScaler.s:VWF)                                 ... 0.21476       7301      16098    0.00058    0.31202 
starbeast2.NodeReheight2(Reheight.t:Species)                     ...       -      98775    1438750    0.03846    0.06424 
starbeast2.CoordinatedUniform(coordinatedUniform.t:Species)      ...       -     354356     415005    0.01923    0.46058 
starbeast2.CoordinatedExponential(coordinatedExponential.t:Speci ... 0.07335     453543     315251    0.01923    0.58994 
beast.base.inference.operator.UpDownOperator(updownAll:Species)  ... 0.66322      74117     233719    0.00769    0.24077 
starbeast2.RealCycle(constPopSizesSwap.Species)                  ... 2.00000      30178     123980    0.00385    0.19576 k = 2
ScaleOperator(constPopSizesScale.Species)                        ... 0.28757      37835     115633    0.00385    0.24653 
ScaleOperator(constPopMeanScale.Species)                         ... 0.44194      16611      34495    0.00128    0.32503 
ScaleOperator(netDiversificationRateScale.t:Species)             ... 0.20702      13692      37872    0.00128    0.26553 
ScaleOperator(ExtinctionFractionScale.t:Species)                 ... 0.11933       5801      19945    0.00064    0.22532 
beast.base.inference.operator.UniformOperator(ExtinctionFraction ...       -      13197      12145    0.00064    0.52076 
SubtreeSlide(bdSubtreeSlide.t:Species)                           ... 0.06648     236813     532106    0.01923    0.30798 
WilsonBalding(bdWilsonBalding.t:Species)                         ...       -        495     770580    0.01923    0.00064 
Exchange(bdWide.t:Species)                                       ...       -       4258     764403    0.01923    0.00554 
Exchange(bdNarrow.t:Species)                                     ...       -      38229     730568    0.01923    0.04973 
Uniform(bdUniformOperator.t:Species)                             ...       -      59042     710796    0.01923    0.07669 
ScaleOperator(bdTreeRootScaler.t:Species)                        ... 0.87763       7185     147379    0.00385    0.04649
ScaleOperator(bdTreeScaler.t:Species)                            ... 0.98773      27551     126232    0.00385    0.17916 

     Tuning: The value of the operator's tuning parameter, or '-' if the operator can't be optimized.
    #accept: The total number of times a proposal by this operator has been accepted.
    #reject: The total number of times a proposal by this operator has been rejected.
      Pr(m): The probability this operator is chosen in a step of the MCMC (i.e. the normalized weight).
  Pr(acc|m): The acceptance probability (#accept as a fraction of the total proposals for this operator).


Total calculation time: 375.843 seconds
End likelihood: -16950.572066013454
Done!\end{verbatim}

\subsection{Checking the log file}
\label{subsec:checkLog}

To check parameters other than species tree topologies and branch values, or to verify
that the chain has been run long enough to reliably represent the posterior distribution,
we will use Tracer. Start the Tracer app and then open the ``starbeast.log'' file.
The first statistic that will be displayed is a histogram of the log posterior probability
(Figure~\ref{fig:tracerPosterior}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/tracerPosterior.png}
\caption
{Opening a log file in Tracer}
\label{fig:tracerPosterior}
\end{figure}

The posterior probability is the sum of the likelihood (which is the sum of
log phylogenetic likelihoods for all sites for all loci), the prior
probability (which is the sum of log prior probabilities for all parameters),
and the speciescoalescent (which is the sum of log coalescent probabilities for
all gene trees). TreeHeight.Species is the height of the root node of the
species tree, and TreeLength.Species is the sum of all branch lengths in the
species tree.

Tracer computes effective sample sizes (ESS) for each logged statistic and
parameter. As a rule, ESS values should be at least 200, particularly for
the important statistics just noted and for parameters of interest. Because of the stochastic
nature of MCMC algorithms, your values \textbf{will be different} to those in
Figure~\ref{fig:tracerPosterior}.

\subsection{Checking the species trees}
\label{subsec:checkTrees}

Start the DensiTree app (included with BEAST2), and then open the
``species.trees'' file. Under the Show panel, enable the Root Canal tree to
get an idea of the most plausible species tree. Then open the Grid panel, and
enable the full grid (Figure~\ref{fig:densitree}). If you want to check the
clade posterior probabilities, select ``View clade toolbar'' from the Window menu.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/densitree.png}
\caption
{Viewing the species trees in DensiTree}
\label{fig:densitree}
\end{figure}

You can see that using a fixed clock rate of $10^{-3}$ substitutions per site
per year, the split between \textit{Canis latrans} (coyotes) and \textit{Canis lupus}
(wolves) is probably less than 500,000 years ago. The age of the most
recent common ancestor (MRCA) of extant \textit{Canis} taxa is approximately 2.5 million
years ago, more recent than the split between humans and chimpanzees \parencite{PradoMartinez2013}.

\subsection{Generating a summary tree}
\label{subsec:makeSummaryTree}

Summary trees are a way of reducing a posterior distribution of tree
topologies and times to a single tree. This is often more readable than the
cloud of trees that DensiTree displays, but researchers should be cautious not
to give it too much emphasis because for most clades and data
sets there is substantial uncertainty in the posterior distribution of
topologies and times.

To generate a summary tree, open the Tree Annotator app included with BEAST2.
Set the burnin percentage to 10 and choose the ``species.trees'' file generated by
StarBEAST2 as the input file. Specify the output file to be something sensible
like ``summary.tree'' and click Run to generate the summary tree
(Figure~\ref{fig:treeAnnotator}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.5\textwidth]{figures/treeAnnotator.png}
\caption
{Running Tree Annotator}
\label{fig:treeAnnotator}
\end{figure}

Load \url{http://icytree.org/} in your web browser and open the summary tree file using the ``Load from file'' option in the File menu. In the Style menu, under ``Node height error bars,'' choose ``CAheight\_95\%\_HPD'' to display 95\% highest posterior density (HPD) intervals of species divergence times on each branch. Also in the Style menu, under ``Internal node text,'' choose ``dmv1'' to display estimates of ancestral effective population size. Enable the time axis by typing the letter ``a'' or by going through the Style menu again (Figure~\ref{fig:icytree}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/icytree.png}
\caption
{Running IcyTree. This tree been made more legible by increasing font size and limiting numbers to two significant figures, further options under IcyTree's ``Style'' menu.}
\label{fig:icytree}
\end{figure}

In StarBEAST2 and many other BEAST packages, effective population sizes are
scaled by generation time. That means if generation times (the average
number of years from zygote to zygote) vary between species in your
analysis, the effective population sizes of different branches cannot be
directly compared.

If the average generation time is 5 years, and our tree is scaled in millions
of years, then the generation time is $5\times10^{-6}$. To get effective
population sizes in numbers of individuals, divide each value by the
generation time. So if the scaled effective population size is 1, that
will correspond to 200,000 individuals.{}

\section{Estimating per-species clock rates}
\label{sec:relaxedClock}

We can estimate molecular clock rates separately for each species using
a relaxed clock model, for example the uncorrelated log-normal (UCLN) model.
First create a new folder for this analysis with a sensible name, something
like ``CanisUCLN''. Relaunch BEAUti, and select ``SpeciesTreeUCLN'' from the
Templates submenu of the File menu (Figure~\ref{fig:speciesTreeUCLN}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/speciesTreeUCLN.png}
\caption
{Starting a UCLN relaxed clock analysis}
\label{fig:speciesTreeUCLN}
\end{figure}

Import the same FASTA files as in
subsection~\ref{subsec:importingAlignments}
(Figure~\ref{fig:fastaFileImport}). For any of the species tree relaxed clock
templates, do \textbf{NOT} link the clock models. This will break the inference
of per-species clock rates.

Assign the same species names as in
subsection~\ref{subsec:speciesNames} (Figure~\ref{fig:guessTaxonMap}, \ref{fig:taxonMap}). This time
we will keep the default setting for the population model (Analytical Integration), so that our analysis will run slightly faster. Set all
the site models to HKY with empirical frequencies as in
subsection~\ref{subsec:siteModel} (Figure~\ref{fig:hky}, \ref{fig:cloneSiteModel}).

Open the Clock Model tab, and manually set the Clock.rate to 0.001 (or
equivalently 1e-3) for \textbf{every} locus (Figure~\ref{fig:uclnClockRates}).
This is necessary because for technical reasons we cannot link the clock
models for the species tree relaxed clock templates.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/uclnClockRates.png}
\caption
{Manually setting every clock rate}
\label{fig:uclnClockRates}
\end{figure}

Open the Priors tab and change the species tree prior from Yule to Birth-Death
to allow for extinction. Finally, open the MCMC tab and change the length of
the chain to 40 million, as in subsection~\ref{subsec:MCMC}
(Figure~\ref{fig:chainLength}). Save the file in the folder you created
previously for this analysis, and give it a sensible name like ``Canis-UCLN.xml''.

Run the MCMC chain by opening the XML file in BEAST, or running it from
the command line as in subsection~\ref{subsec:runningBEAST}. This will
take about twice as long (10 to 20 minutes) as the fixed clock analysis,
because relaxed clocks are more computationally intensive.

After BEAST has finished, open the ``starbeast.log'' file in Tracer to
check that the important statistics have ESS values of at least 200.
Select the branchRatesStdev.Species parameter (Figure~\ref{fig:tracerUCLN}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/tracerUCLN.png}
\caption
{Checking the species tree relaxed clock analysis in Tracer}
\label{fig:tracerUCLN}
\end{figure}

This parameter models the spread of molecular clock rates among branches in
the species tree. The default prior for this parameter in StarBEAST2 is a
log-normal distribution with a mean of 1, but the posterior distribution of this
parameter has a mean of about 0.2. This means that the data is pulling this
parameter lower, probably because there is very little variation in clock
rates between species. Indeed the mode of the posterior distribution is about
zero, suggesting that a strict clock may be more appropriate
(Figure~\ref{fig:tracerUCLN}).{}

\section{Total evidence tip-dating}
\label{sec:FBD}

StarBEAST is able to combine
tip-dating using the fossilized birth death (FBD) process with multispecies
coalescent (MSC) inference of species and gene trees. We call this integrative
model ``FBD-MSC''. To begin your analysis, creat a new folder for this
purpose with a sensible name, something like ``CanisFBD''.

\textbf{When setting up an FBD-MSC analysis in BEAUti, it is necessary to
import morphological data \ul{after} specifying the taxon sets \ul{and} the
tree model, but \ul{before} specifying the tip dates.} The following steps
will be consistent with that ordering. First, repeat
section~\ref{sec:speciesTree} from subsection~\ref{subsec:importingAlignments}
to and including \ref{subsec:speciesNames}. Leave the population model set at
the default (Analytical Integration) to save time, then set
all site models to HKY with empirical frequencies as in
subsection~\ref{subsec:siteModel}.

Before adding the morphological data, go to the Priors panel and change
the prior on the species tree from Yule to FBD
(Figure~\ref{fig:changeTreePrior}). This is necessary because the inclusion of
fossil taxa means our species tree will be \textit{serially sampled} rather
than \textit{ultrametric}. StarBEAST2 gives you the option of various
birth-death models -- Yule, Calibrated Yule, Birth Death and FBD -- but only
FBD is valid for serially sampled trees.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/changeTreePrior.png}
\caption
{Change the tree model from Yule to FBD.}
\label{fig:changeTreePrior}
\end{figure}

Now go back to the Partitions tab, and from the File menu select
``Add Morphology to Species Tree'' (Figure~\ref{fig:addMorphology}).
Navigate to the data folder of the tutorial and import the
``morphology\_canis.nex'' file.\footnote{This file contains recorded states for 50
morphological characters of \textit{Canis} species and some related species,
taken from \cite{Slater2015}.}

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/addMorphology.png}
\caption
{The option to add morphological data to the species tree}
\label{fig:addMorphology}
\end{figure}

\clearpage

Select Yes to condition on recording variable characters only (Mkv). Mkv is
used when only characters that vary between species have been included in the
character matrix, as is the case for this tutorial's data set. You
should now see four morphology partitions
(Figure~\ref{fig:morphologyPartitions}), one for each number of character
states. For example, morphology\_canis2 is for binary characters.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/morphologyPartitions.png}
\caption
{After morphological data has been added to the species tree}
\label{fig:morphologyPartitions}
\end{figure}

Open the Tip Dates panel and enable ``Use tip dates''. Change
``Since some time in the past'' to ``before the present''. Click
on Auto-configure and select ``read from file''. Choose the
``tip\_dates\_canis.dat'' file in the data folder of the tutorial.
Click OK, and your tip dates should look like Figure~\ref{fig:tipDates}.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/tipDates.png}
\caption
{Configured tip dates}
\label{fig:tipDates}
\end{figure}

Go to the Clock Model panel, and enable ``estimate'' for the
molecular clock (by default called APOBS1 after the first locus, although
this can be changed in the Partitions tab). Then select the morphological
data partition and enable ``estimate'' so that the morphological
clock rate will also be estimated (Figure~\ref{fig:estimateMorphClock}).
Make sure you have enabled ``estimate'' for \textbf{both} clocks, since we
are using fossil data to calibrate our tree.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/estimateMorphClock.png}
\caption
{Estimating the clock rates}
\label{fig:estimateMorphClock}
\end{figure}

Next, go back to the Priors panel, scroll down to the strictClockRate.c:molecular
parameter, and expand that prior. Change the mean to 0.001, the \textit{a
priori} molecular clock rate estimate from section~\ref{sec:speciesTree}.
Leave the standard deviation set at 1, a moderately informative prior that
will still allow the rate to be guided by the fossil data. Then change the
prior distribution for the strictClockRate.c:morphology\_canis prior to 1/X
(Figure~\ref{fig:clockPriors}). This is an improper, uninformative prior so
the inferred morphological clock rate will be estimated solely from the data.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/clockPriors.png}
\caption
{Configuring the clock prior distributions}
\label{fig:clockPriors}
\end{figure}

\clearpage

The FBD-MSC model is quite complex and requires very long chains for
reliable sampling. Open the MCMC tab and change the chain length to 200
million. We will need to reduce the sampling rate so that the output files remain
managably small, so change the Store Every value to 50,000. Expand the
tracelog, speciesTreeLogger, screenlog and every gene tree log, and set the
Log Every value to 50,000 for each of them. This will limit the number of
samples to 4,000 in each log file (Figure~\ref{fig:samplingRate}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/samplingRate.png}
\caption
{Changing the sampling rates after settting a chain length of 200 million}
\label{fig:samplingRate}
\end{figure}

Save the XML file in the folder you created for this analysis, and name it
something like ``CanisFBD.xml''. Run the MCMC chain by opening the XML file in
BEAST, or running it from the command line as in
subsection~\ref{subsec:runningBEAST}. The chain should take between 1 and 2 hours to
finish, depending on the speed of your computer.

\begin{tcolorbox}[colback=orange!5,colframe=orange!40!black,title=Editing BEAST XML files]
BEAUti is a wonderful tool, but sometimes it can be faster to edit XML files directly rather than using BEAUti to configure every aspect of your model. Setting the sampling rate for gene tree log files one at a time is a pain, especially if you have 100 or more loci in your analysis!

Instead of changing the sampling rates in BEAUti, you can leave them at their default values, and use a text editor to change them all at once. After opening your XML file in the editor, find-and-replace all occurences of the sampling rate attribute and its default value, e.g. \texttt{logEvery="5000"}, with the value changed to the one desired, e.g. \texttt{logEvery="50000"}.

You may even want to design a model in a way that is impossible to configure with BEAUti, in which case you \textit{must} edit the XML file directly. However, that sort of thing is beyond the scope of this tutorial.
\end{tcolorbox}

After BEAST has finished, open the ``starbeast.log'' file in Tracer to check
on the ESS values. These values are likely insufficient given a chain length
of only 200 million for such a complex model, and a real analysis might require
around 2 billion states, possibly spread across multiple chains.

Select the strictClockRate.c:molecular parameter
(Figure~\ref{fig:clockRateHistogram}). The 95\% HPD interval for this parameter
spans approximately 0.0005 to 0.0013 substitutions per site per million years,
in agreement with the \textit{a priori} rate of 0.001.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/clockRateHistogram.png}
\caption
{The estimated posterior distribution of the molecular clock rate}
\label{fig:clockRateHistogram}
\end{figure}

Now open the ``Trace'' tab to see how this value wandered around over the
course of the MCMC chain. With only 200 million states this is unlikely
to look like the ideal ``hairy catepillar,'' and will instead exhibit slow
transitions between modes of the distribution (Figure~\ref{fig:clockRateTrace}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/clockRateTrace.png}
\caption
{The MCMC trace of the molecular clock rate}
\label{fig:clockRateTrace}
\end{figure}

\clearpage

Start the DensiTree app again, and then open the ``species.trees'' file. Under
the Show panel, enable the Root Canal tree to get an idea of the most
plausible species tree. Then open the Grid panel, and enable the full grid.
Compared with the fixed clock analysis (Figure~\ref{fig:densitree}), the
divergence time between \textit{Canis lupus} and \textit{Canis latrans} is a
little older, and the age of the MRCA of extant species is a little younger
(Figure~\ref{fig:sampledAncestorsDensitree}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/sampledAncestorsDensitree.png}
\caption
{Cloud of FBD-MSC species trees}
\label{fig:sampledAncestorsDensitree}
\end{figure}

Open the Tree Annotator app included with BEAST2, and set the burnin
percentage to 10. Trees inferred by the FBD-MSC model as implemented in
StarBEAST2 can include sampled ancestors \parencite{Gavryushkina2014}, which are
incompatible with Common Ancestor heights \parencite{Heled2013} as implemented in
Tree Annotator. So in order to generate a summary tree, change ``Node heights'' to
``Mean heights'' (Figure~\ref{fig:treeAnnotatorMeanHeights}). Choose the
``species.trees'' file as the input file, and specify the output file to be
something sensible like ``summary.tree'', then click Run.

\begin{figure}[htb!]
\centering
\includegraphics[width=0.5\textwidth]{figures/treeAnnotatorMeanHeights.png}
\caption
{Running Tree Annotator for sampled ancestor trees}
\label{fig:treeAnnotatorMeanHeights}
\end{figure}

Go back to IcyTree (\url{http://icytree.org/}) in your web browser and open
the summary tree file using the ``Load from file'' option in the File menu.
In the Style menu, under ``Node height error bars,'' choose
``height\_95\%\_HPD'' to display 95\% highest posterior density
(HPD) intervals of species divergence times on each branch. This time show
the posterior support of each clade by choosing ``posterior'' in ``Internal node text''
from the Style menu. Enable the time axis again through the Style menu or by typing the
letter ``a'' (Figure~\ref{fig:sampledAncestorsIcyTree}).

\begin{figure}[htb!]
\centering
\includegraphics[width=0.8\textwidth]{figures/sampledAncestorsIcyTree.png}
\caption
{Showing posterior support for clades in IcyTree}
\label{fig:sampledAncestorsIcyTree}
\end{figure}

The tree topology is similar to the fixed clock analysis
(Figure~\ref{fig:icytree}), except that \textit{Cuon alpinus} and
\textit{Lycaon pictus} are now a clade with 100\% support. A disagreement
between molecular data which contradicts this clade and morphological
data which supports it has been reported previously \parencite{Zrzavy2004}. This shows
how morphological characters and fossil data can influence species tree
estimates.

\section{Wrapping up}

That concludes the tutorial. One final note; when runnning any MCMC method, it
is possible than the chain has gotten stuck in a region of parameter space
with high local probability. So for any kind of analysis intended for
publication, it is a good idea to run at least two MCMC chains. Open both
chains in Tracer, and check that the traces for the logged statistics and
parameters have the same distributions.

\clearpage

\printbibliography

\end{document}
